pkg/services/dashboards/service/dashboard_service.go

```diff
@@ -1715,10 +1724,54 @@ func (dr *DashboardServiceImpl) searchDashboardsThroughK8sRaw(ctx context.Contex
 		request.Options.Fields = append(request.Options.Fields, req...)
 	}
 
+	if query.IsDeleted {
+		request.IsDeleted = query.IsDeleted
+	}
+
 	if query.Limit > 0 {
 		request.Limit = query.Limit
 	}
 
+	if query.Page > 0 {
```

Comment:
To match the current behavior: https://github.com/grafana/grafana/blob/main/pkg/services/dashboards/database/database.go#L968

Without this, if you do /api/search?page=4, it will return the full list of everything (which is different than the current behavior)

```suggestion

	if query.Limit > 0 {
		request.Limit = query.Limit
	} else {
		request.Limit = 1000	
	}

	if query.Page > 0 {
```